src/darker/tests/test_command_line.py

```diff
@@ -529,14 +546,101 @@ def test_black_options(monkeypatch, tmpdir, git_repo, options, expect):
         {"main.py": 'print("Hello World!")\n'}, commit="Initial commit"
     )
     added_files["main.py"].write_bytes(b'print ("Hello World!")\n')
-    with patch.object(black_diff, "Mode", wraps=black_diff.Mode) as file_mode_class:
+    with patch.object(
+        black_formatter, "Mode", wraps=black_formatter.Mode
+    ) as file_mode_class:
 
         main(options + [str(path) for path in added_files.values()])
 
     _, expect_args, expect_kwargs = expect
     file_mode_class.assert_called_once_with(*expect_args, **expect_kwargs)
 
 
+@pytest.mark.kwparametrize(
+    dict(options=[]),
+    dict(options=["-c", "ruff.cfg"], expect_opts=["--line-length=81"]),
+    dict(options=["--config", "ruff.cfg"], expect_opts=["--line-length=81"]),
+    dict(
+        options=["-S"],
+        expect_opts=['--config=format.quote-style="preserve"'],
+    ),
+    dict(
+        options=["--skip-string-normalization"],
+        expect_opts=['--config=format.quote-style="preserve"'],
+    ),
+    dict(options=["-l", "90"], expect_opts=["--line-length=90"]),
+    dict(options=["--line-length", "90"], expect_opts=["--line-length=90"]),
+    dict(
+        options=["-c", "ruff.cfg", "-S"],
+        expect_opts=["--line-length=81", '--config=format.quote-style="preserve"'],
+    ),
+    dict(
+        options=["-c", "ruff.cfg", "-l", "90"],
+        expect_opts=["--line-length=90"],
+    ),
+    dict(
+        options=["-l", "90", "-S"],
+        expect_opts=["--line-length=90", '--config=format.quote-style="preserve"'],
+    ),
+    dict(
+        options=["-c", "ruff.cfg", "-l", "90", "-S"],
+        expect_opts=["--line-length=90", '--config=format.quote-style="preserve"'],
+    ),
+    dict(options=["-t", "py39"], expect_opts=["--target-version=py39"]),
+    dict(options=["--target-version", "py39"], expect_opts=["--target-version=py39"]),
+    dict(
+        options=["-c", "ruff.cfg", "-t", "py39"],
+        expect_opts=["--line-length=81", "--target-version=py39"],
+    ),
+    dict(
+        options=["-t", "py39", "-S"],
+        expect_opts=[
+            "--target-version=py39",
+            '--config=format.quote-style="preserve"',
+        ],
+    ),
+    dict(
+        options=["-c", "ruff.cfg", "-t", "py39", "-S"],
+        expect_opts=[
+            "--line-length=81",
+            "--target-version=py39",
+            '--config=format.quote-style="preserve"',
+        ],
+    ),
+    dict(options=["--preview"], expect_opts=["--preview"]),
+    expect_opts=[],
+)
+def test_ruff_options(monkeypatch, tmpdir, git_repo, capsys, options, expect_opts):
+    """Ruff options from the command line are passed correctly to Ruff"""
+    monkeypatch.chdir(tmpdir)
+    (tmpdir / "pyproject.toml").write("[tool.ruff]\n")
```

Comment:
os.path.join(tmpdir, "pyproject.toml") os agnostic

Reply:
I was under the impression that `py.path.local() / "subpath"` using the Pytest [tmpdir] fixture is also OS agnostic (see [Working with Paths] in Pytest `py.path` documentation). I think I've used it successfully elsewhere without Windows issues. Any gotchas I'm not aware of?

But, this is a good reminder to follow Pytest's [recommendation] here, too, to migrate to the [tmp_path] fixture which returns a [pathlib.Path]! I'm pretty sure `pathlib.Path() / "subpath"` is OS agnostic (see [Operators] in `pathlib` documentation). 

[tmpdir]: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture
[Working with Paths]: https://py.readthedocs.io/en/latest/path.html#working-with-paths
[recommendation]: https://py.readthedocs.io/en/latest/path.html#py-path
[tmp_path]: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture
[pathlib.Path]: https://docs.python.org/3/library/pathlib.html#pathlib.Path
[Operators]: https://docs.python.org/3/library/pathlib.html#operators